addons/account/models/res_partner_bank.py

```diff
@@ -281,14 +281,19 @@ def write(self, vals):
         # leaves them vulnerable to edits via the shell/... So we need to ensure that the user has the rights to edit
         # these fields when writing too.
         # While we do lock changes if the account is trusted, we still want to allow to change them if we go from not trusted -> trusted or from trusted -> not trusted.
-        any_trusted_accounts = any(account.lock_trust_fields for account in self)
-        if not any_trusted_accounts:
+        trusted_accounts = self.filtered(lambda x: x.lock_trust_fields)
+        if not trusted_accounts:
             should_allow_changes = True  # If we were on a non-trusted account, we will allow to change (setting/... one last time before trusting)
         else:
             # If we were on a trusted account, we only allow changes if the account is moving to untrusted.
             should_allow_changes = ('allow_out_payment' in vals and vals['allow_out_payment'] is False)
 
-        if ('acc_number' in vals or 'partner_id' in vals) and not should_allow_changes:
+        lock_fields = {'acc_number', 'sanitized_acc_number', 'partner_id', 'acc_type'}
+        if not should_allow_changes and any(
+            account[fname] != account.new({fname: vals[fname]})[fname]
```

Comment:
sorry I find it doesn't work for `many2one` 😢  `partner_id`
the right will be record with `NewId` which is not equal to the left

Reply:
did you check changing `partner_id` and make sure this line is executed for `partner_id`? The normal record (left) should always be not equal to the newid record (right) I think.

Reply:
I did, I made sure to check that it failed when needed and not if it shouldn't fail. It was never a NewId anywhere.

Reply:
we can aslo implement this logic by saving these record values before and comparing them after the `super().write`
Although I proposed this strategy and it is safe for the listed fields. I am worried that other developers may abusively use the `new` api like this for delgeated many2one or x2omany fields where the right part will be newId records

@rco-odoo what do you think?

Reply:
The code is actually fine for many2one fields except _delegate_ fields (the ones in `_inherits`).  We currently don't have an API to know whether the field will actually be modified, but it should come someday.  What the new record does is convert the value to the _cache_ format, then convert it to the _record_ format.  You could write it that way:
```suggestion
            account[fname] != account._fields[fname].convert_to_record(
                account._fields[fname].convert_to_cache(vals[fname], account),
                account,
            )
```
You may factor out `account._fields[fname]` as `field` below...

Reply:
@rco-odoo That is alright in the ORM, but it's too complex for business code...

Could I add a helper in the ORM?

Reply:
Done.